python/cudf/cudf/_libxx/join.pyx

```diff
@@ -111,6 +112,7 @@ cpdef join(Table lhs,
         for name in right_on:
             right_on_ind.push_back(right_join_cols.index(name))
 
+    c_columns_in_common = list(columns_in_common)
```

Comment:
Does the order of this matter? Using a set will make the order an implementation detail / undefined.

Reply:
I thought not but the answer appears to be yes and this could lead to mislabling the column names. If we need something that retains order but has constant time complexity for it's `setitem` would another odict be preferable here then?

Reply:
Yea seems like an OrderedDict would be appropriate here then 😄

Reply:
Ok - thank you for confirming. I had some trouble making sure that an odict doesn't call append on some hidden data structure under the hood. Will update this.